libvast/src/system/pivot_command.cpp

```diff
@@ -56,14 +51,17 @@ caf::message pivot_command(const invocation& inv, caf::actor_system& sys) {
     = spawn_or_connect_to_node(self, inv.options, content(sys.config()));
   if (auto err = std::get_if<caf::error>(&node_opt))
     return caf::make_message(std::move(*err));
-  const auto& node = std::holds_alternative<node_actor>(node_opt)
-                       ? std::get<node_actor>(node_opt)
-                       : std::get<scope_linked<node_actor>>(node_opt).get();
+  auto local_node = !std::holds_alternative<node_actor>(node_opt);
+  const auto& node = local_node
+                       ? std::get<scope_linked<node_actor>>(node_opt).get()
+                       : std::get<node_actor>(node_opt);
   VAST_ASSERT(node != nullptr);
-  // Start signal monitor.
-  std::thread sig_mon_thread;
-  auto guard = system::signal_monitor::run_guarded(
-    sig_mon_thread, sys, defaults::system::signal_monitoring_interval, self);
+  if (local_node) {
+    // Register as the termination handler.
+    auto signal_reflector
+      = sys.registry().get<signal_reflector_actor>("signal-reflector");
+    self->send(signal_reflector, atom::subscribe_v);
+  }
```

Comment:
None of these commands used the signal monitor before, so I did not "forget" about them. 
I guess it should be fine to do this in `spawn_node`.

Reply:
Fair enough, I didn't actually grep for that but for `spawn_or_connect_to_node` instead. Still makes sense to tie this together, and also to change `flush` to use `connect_to_node` instead.

Reply:
I moved the `subscribe` part to `spawn_node` and added handling to `show`.
I think `rebuild stop` should also just use `connect_to_node`. `rebuild start` should work with `-N`, but it looks like it currently doesn't.
`web server` is a bit more involved because the main loop is managed by restinio.

Reply:
`rebuild -N start` has the problem that the catalog returns an empty set before it received the partition synopses from the index. I hope that we can soon move the loading logic into the catalog to resolve this properly. I think it's fine to leave this with `spawn_or_connect_to_node`.